source/cppexpose/include/cppexpose/typed/TypedArray.inl

```diff
@@ -24,13 +24,47 @@ TypedArray<T, ET, Size, BASE>::TypedArray()
 {
 }
 
+
+template <typename T, typename ET, size_t Size, typename BASE>
+TypedArray<T, ET, Size, BASE>::TypedArray(const TypedArray & rhs)
+: Typed<T, BASE>(rhs)
+{
+    // don't copy m_subValues which will be lazy-initialized when used
+}
+
+
+template <typename T, typename ET, size_t Size, typename BASE>
+TypedArray<T, ET, Size, BASE>::TypedArray(TypedArray && rhs)
+: Typed<T, BASE>(std::move(rhs))
+, m_subValues(std::move(rhs.m_subValues))
+{
+}
+
 template <typename T, typename ET, size_t Size, typename BASE>
 TypedArray<T, ET, Size, BASE>::~TypedArray()
 {
-    for (AbstractTyped * subValue : m_subValues)
-    {
-        delete subValue;
-    }
+}
+
+
+template <typename T, typename ET, size_t Size, typename BASE>
+TypedArray<T, ET, Size, BASE> & TypedArray<T, ET, Size, BASE>::operator=(const TypedArray & rhs)
+{
+    Typed<T, BASE>::operator=(std::move(rhs));
+
+    // don't copy m_subValues which will be lazy-initialized when used
```

Comment:
I think here is a different behavior implemented than the one in the constructor.
This behavior here initializes ```m_subValues``` through the ```operator=(TypedArray &&)``` call but this comment states otherwise.